### PR TITLE
PublicAPI: Add GET /me/media

### PR DESCRIPTION
**Pull request by InnayTool**
Route that returns all uploads made by the authenticated user.

### Component/Part
Public

### Description
Route that returns all uploads made by the authenticated user.

### Related Issue(s)
#1009

### CODE REVIEW

File: src/media/media-upload.dto.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  link: string;
+
+  @IsString()
+  noteId: string;
+
+  @IsDate()
+  createdAt: Date;
+
+  @IsString()
+  userName: string;
```
Comment (DerMolly) [inline src/media/media-upload.dto.ts:20]: These should get JSDoc-Strings like the other Dtos…

File: src/media/media.service.spec.ts [hunk 2]
```diff
@@ -228,4 +228,29 @@ describe('MediaService', () => {
       }
     });
   });
+
+  describe('listUploadsByUser', () => {
+    it('works with one upload from user', async () => {
```
Comment (DerMolly) [inline src/media/media.service.spec.ts:233]: One could move both `it`s into a `describe` with the title `works`

File: src/media/media.service.ts [hunk 3]
```diff
@@ -157,6 +159,23 @@ export class MediaService {
...
+   * @param {User} user - the specific user
+   * @return {MediaUpload[]} arary of media uploads owned by the user
+   */
+  async listUploadsByUser(user: User): Promise<MediaUpload[]> {
+    const mediaUploads = await this.mediaUploadRepository.find({
+      where: { user: user },
+      relations: ['user', 'note'],
+    });
+    if (mediaUploads === undefined) {
+      return [];
```
Comment (DerMolly) [inline src/media/media.service.ts:174]: Codecov want's that this line is also covered by either an e2e or unit test.

**Review by DerMolly — CHANGES_REQUESTED** [review]

File: src/media/media.service.spec.ts (hunk 2 above)
Comment (ErikMichelson) [reply src/media/media.service.spec.ts:233]: IMHO that only makes sense if you also have a describe block and tests for "works not".

File: src/media/media-upload.dto.ts (hunk 1 above)
Comment (InnayTool) [reply src/media/media-upload.dto.ts]: fixed

File: src/media/media.service.spec.ts (hunk 2 above)
Comment (InnayTool) [reply src/media/media.service.spec.ts]: I changed it.

File: src/media/media.service.ts (hunk 3 above)
Comment (InnayTool) [reply src/media/media.service.ts:174]: Added another test for that to `media.service.spec.ts`.

**Review by DerMolly — APPROVED** [review]

File: docs/content/dev/public_api.yml [hunk 4]
```diff
@@ -900,19 +900,21 @@ components:
...
       type: object
       properties:
-        note:
-          type: string
-          description: ID of the note the file was uploaded to
-        user:
-          type: string
-          description: username of the user who uploaded the file
-        url:
+        link:
```
Comment (davidmehren) [inline docs/content/dev/public_api.yml:903]: Why the change from `url` to `link`?
I find "url" to be a way more descriptive term than "link".

File: docs/content/dev/public_api.yml [hunk 5]
```diff
@@ -900,19 +900,21 @@ components:
...
           type: string
           description: URL of the file
+        noteId:
```
Comment (davidmehren) [inline docs/content/dev/public_api.yml:906]: Maybe change this to `owningNote` or `parentNote` or similar?
That would make the object more self-documenting.

File: docs/content/dev/public_api.yml [hunk 6]
```diff
@@ -900,19 +900,21 @@ components:
...
+          type: string
+          description: ID of the note the file was uploaded to
         createdAt:
           type: string
           format: date-time
           description: Date when the file was upladed
+        userName:
```
Comment (davidmehren) [inline docs/content/dev/public_api.yml:913]: Same as above

File: src/media/media-upload.dto.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { IsDate, IsString } from 'class-validator';
+
+export class MediaUploadDto {
+  /**
+   * The link to the media file.
+   * @example "example.com/uploads/testfile123.jpg"
```
Comment (davidmehren) [inline src/media/media-upload.dto.ts:12]: Does the property really not contain the protocol?

File: src/media/media.service.spec.ts [hunk 8]
```diff
@@ -228,4 +228,39 @@ describe('MediaService', () => {
...
+          backendData: 'testBackendData',
+          user: {
+            userName: 'hardcoded',
+          } as User,
+        } as MediaUpload;
+        jest
+          .spyOn(mediaRepo, 'find')
+          .mockResolvedValueOnce([mockMediaUploadEntry]);
+        expect(
+          await service.listUploadsByUser({ userName: 'hardocded' } as User),
```
Comment (davidmehren) [inline src/media/media.service.spec.ts:246]: Typo in `userName`

File: src/media/media.service.spec.ts [hunk 9]
```diff
@@ -228,4 +228,39 @@ describe('MediaService', () => {
...
+        ).toEqual([mockMediaUploadEntry]);
+      });
+
+      it('without uploads from user', async () => {
+        jest.spyOn(mediaRepo, 'find').mockResolvedValueOnce([]);
+        const mediaList = await service.listUploadsByUser({
+          userName: 'hardocded',
```
Comment (davidmehren) [inline src/media/media.service.spec.ts:253]: typo

File: src/media/media.service.spec.ts [hunk 10]
```diff
@@ -228,4 +228,39 @@ describe('MediaService', () => {
...
+        } as User);
+        expect(mediaList).toEqual([]);
+      });
+      it('with error (undefined as return value of find)', async () => {
+        jest.spyOn(mediaRepo, 'find').mockResolvedValueOnce(undefined);
+        const mediaList = await service.listUploadsByUser({
+          userName: 'hardocded',
```
Comment (davidmehren) [inline src/media/media.service.spec.ts:260]: typo

**Review by davidmehren — CHANGES_REQUESTED** [review]

**Review by davidmehren — APPROVED** [review]

**Review by DerMolly — APPROVED** [review]
